ipamutils/utils_test.go

```diff
@@ -1,16 +1,36 @@
 package ipamutils
 
 import (
-	"testing"
-
 	_ "github.com/docker/libnetwork/testutils"
+	"github.com/stretchr/testify/assert"
+	"testing"
 )
 
 func init() {
-	InitNetworks()
+	prePool1 := &PredefinedPools{"172.80.0.0/16", 24}
+	prePool2 := &PredefinedPools{"172.90.0.0/16", 24}
+	DefaultAddressPools := []*PredefinedPools{prePool1, prePool2}
+	InitNetworks(DefaultAddressPools)
+}
+
+func TestInitAddressPools(t *testing.T) {
+
+	// Check for Random IPAddresses in PredefinedBroadNetworks  ex: first , last and middle
+	if len(PredefinedBroadNetworks) == 512 {
+		assert.Equal(t, PredefinedBroadNetworks[0].String(), "172.80.0.0/24")
+		assert.Equal(t, PredefinedBroadNetworks[127].String(), "172.80.127.0/24")
+		assert.Equal(t, PredefinedBroadNetworks[255].String(), "172.80.255.0/24")
+		assert.Equal(t, PredefinedBroadNetworks[256].String(), "172.90.0.0/24")
+		assert.Equal(t, PredefinedBroadNetworks[383].String(), "172.90.127.0/24")
+		assert.Equal(t, PredefinedBroadNetworks[511].String(), "172.90.255.0/24")
+
+	} else {
+		t.Fatal(" Failed to find PredefinedBroadNetworks")
+	}
 }
 
 func TestGranularPredefined(t *testing.T) {
+	PredefinedBroadNetworks = nil
```

Comment:
Why is this change needed ?

Reply:
1) Sync.Once is taken by modifying whole code base to make sure we call it only once.

2)  func TestGranularPredefined(t *testing.T) {
+	PredefinedBroadNetworks = nil
This test case was failing due to subnet  exists issue with our newly added change . so before starting the test case I make sure it is set to nil and run the test cases.